src/functional/iterator/embedded.py

```diff
@@ -26,6 +26,11 @@ def deref(it):
     return it.deref()
 
 
+@builtins.can_deref.register(EMBEDDED)
+def can_deref(it):
+    return not it.is_none()  # TODO refactor is_none to can_deref in iterator
```

Comment:
Note that an implementation of `.is_none()` is only available for MDIterator. Other iterators (`wrap_iterator` in `lift`, `ScanArgIterator`) are missing this method.

Reply:
replaced is_none by can_deref and added to all iterators.